tutorials/horizontally-scale-mysql-database-backend-with-google-cloud-sql-and-proxysql/index.md

```diff
@@ -0,0 +1,325 @@
+
+## Configuring ProxySQL to Distribute Database Queries
+
+### Adding Read Replicas
+
+In the ProxySQL admin interface, run the following SQL queries:
+
+```
+INSERT INTO mysql_servers (hostgroup_id, hostname) VALUES ('1', '[IP_ADDRESS_#1]'), ('1', '[IP_ADDRESS_#2]')
+LOAD MYSQL SERVERS TO RUNTIME
+SAVE MYSQL SERVERS TO DISK
+```
+
+Replace `[IP_ADDRESS_#1]` and `[IP_ADDRESS_#2]` with the IPv4 addresses of the two read replicas you just created. Note that the replicas are assigned to a different group (`1`) from the master instance (`0`).
+
+### Updating Rules for Distribution
+
+Note: Steps below set a simple list of rules that distribute all write traffic and 20% read traffic to the master instance, 40% read traffic to the first read replica and another 40% read traffic to the second read replica. It is based on [the sample rules in ProxySQL wiki](https://github.com/sysown/proxysql/wiki/ProxySQL-Read-Write-Split-(HOWTO)#basic-readwrite-split-using-regex). **Those rules are for demonstration purposes only and may not fit your application well.** Allocate different weights to read replicas as you fit and experiment with other query routing functionalities [(regex, digest, etc.)](https://github.com/sysown/proxysql/wiki/ProxySQL-Read-Write-Split-(HOWTO)#readwrite-split-using-regex-and-digest) provided in ProxySQL to create context-sensitive rules for better query distribution.
+
+1. Declare rules with the following SQL queries:
+
+	```
+	INSERT INTO mysql_rules (rule_id, active, match_digest, destination_hostgroup, apply) VALUES (1,1,'^SELECT.*FOR UPDATE',0,1), (2,1,'^SELECT',1,1);
+	LOAD MYSQL RULES TO RUNTIME;
+	SAVE MYSQL RULES TO DISK;
+	```
+
+	The queries add two rules:
+
+	- Distribute all the `SELECT … FOR UPDATE` queries to database servers in host group `1`
+	- Distribute all the `SELECT` queries to database servers in host group `0`
+
+	Queries that do not match any existing rules go to database servers in host group `0` (default).
+
+2. Add the master instance to host group `1` so that it also helps serve read traffic:
+
+	```
+	INSERT INTO mysql_servers (hostgroup_id, hostname) VALUES (1, '[IP_ADDRESS]');
+	LOAD MYSQL SERVERS TO RUNTIME;
+	SAVE MYSQL SERVERS TO DISK;
+	```
+
+3. Adjust the weights of servers in host group `1`:
+
+	```
+	UPDATE mysql_servers SET weight = 40 WHERE hostname = '[IP_ADDRESS_1]';
+	UPDATE mysql_servers SET weight = 40 WHERE hostname = '[IP_ADDRESS_2]';
+	UPDATE mysql_servers SET weight = 20 WHERE hostname = '[IP_ADDRESS]';
+	LOAD MYSQL SERVERS TO RUNTIME;
+	SAVE MYSQL SERVERS TO DISK;
+	```
+
+	Replace `[IP_ADDRESS_1]`, `[IP_ADDRESS_2]`, `[IP_ADDRESS]` with the ip addresses of read replica #1, read replica #2, and the master instance respectively. The weights are calculated as follows:
```

Comment:
"ip addresses" -> "IP addresses"

Reply:
You capitalize "IP" elsewhere in the doc, so you may want to do that here as well for consistency.